.github/workflows/main_ci.yml

```diff
@@ -34,226 +28,116 @@ jobs:
-    - name: Restore Cache
-      uses: actions/cache@v3
-      with:
-        path: ${{ github.workspace }}/build/cache
-        key: ${{ env.CACHE_VERSION }}-${{ env.CACHE_NAME }}-ubuntu-latest-${{ hashFiles('vcpkg_commit.txt', 'vcpkg.json', 'alternatives/openssl_3/vcpkg.json') }}
-        restore-keys: |
-          ${{ env.CACHE_VERSION }}-${{ env.CACHE_NAME }}-ubuntu-latest
-
-    - name: Dependencies
-      run: |
-        sudo apt-get install -y linux-headers-$(uname -r) nasm
-
-    - name: Restore cache
-      uses: actions/cache@v3
-      with:
-        path: ${{ github.workspace }}/build/cache
-        key: VCPKG-BinaryCache-${{ runner.os }}
-
-    - name: Build (OpenSSL 1.1)
-      run: |
-        cmake -B "${{ env.CMAKE_BUILD_DIR }}" -DTESTING=ON -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_TOOLCHAIN_FILE }}"
-        cmake --build "${{ env.CMAKE_BUILD_DIR }}" --target all
-
-    - name: Unit Test (OpenSSL 1.1)
-      run: |
-        cmake --build "${{ env.CMAKE_BUILD_DIR }}" --target test
-
-    - name: Build (Interop Harness)
-      run: |
-        cd cmd/interop
-        cmake -B build -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_TOOLCHAIN_FILE }}"
-        cmake --build build
-
-    - name: Test self-interop
-      run: |
-        make -C cmd/interop self-test
-
-    - name: Test interop on test vectors
-      run: |
-        make -C cmd/interop interop-test
-
-    - name: Test gRPC live interop with self
-      run: |
-        cd cmd/interop
-        ./grpc-self-test.sh
-        
-    - name: Build (OpenSSL 3)
-      run: |
-        cmake -B "${{ env.CMAKE_BUILD_OPENSSL3_DIR }}" -DTESTING=ON -DVCPKG_MANIFEST_DIR="alternatives/openssl_3" -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_TOOLCHAIN_FILE }}"
-        cmake --build "${{ env.CMAKE_BUILD_OPENSSL3_DIR }}"
-
-    - name: Unit Test (OpenSSL 3)
-      run: |
-        cmake --build "${{ env.CMAKE_BUILD_OPENSSL3_DIR }}" --target test
-        
-    - name: Build (BoringSSL)
-      run: |
-        cmake -B "${{ env.CMAKE_BUILD_BORINGSSL_DIR }}" -DTESTING=ON -DVCPKG_MANIFEST_DIR="alternatives/boringssl_1.1" -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_TOOLCHAIN_FILE }}"
-        cmake --build "${{ env.CMAKE_BUILD_BORINGSSL_DIR }}"
-
-    - name: Unit Test (BoringSSL)
-      run: |
-        cmake --build "${{ env.CMAKE_BUILD_BORINGSSL_DIR }}" --target test
-
-  platform-sanitizer-tests:
-    if: github.event.pull_request.draft == false
-    needs: quick-linux-interop-check
-    name: Build and test platforms using sanitizers and clang-tidy
+    name: Build and test
     runs-on: ${{ matrix.os }}
     strategy:
-      fail-fast: false
       matrix:
         os: [windows-latest, ubuntu-latest, macos-latest]
+        crypto: [openssl_1.1, openssl_3, boringssl]
         include:
             - os: windows-latest
-              ossl3-vcpkg-dir: "alternatives\\openssl_3"
-              boringssl-vcpkg-dir: "alternatives\\boringssl_1.1"
               ctest-target: RUN_TESTS
```

Comment:
Even thought it works, I am not a bit fan us using the matrix to hold variables.  I think it is confusing.  I think there should be a top level variable... `CMAKE_TEST_TARGET_DEFAULT` and `CMAKE_TEST_TARGET_WINDOWS` - Yes, it would be an conditional step, but you would be able to remove the confusing include section of the matrix.

Reply:
Good point, will do.